lib/Dialect/Triton/IR/DiscardableAttributes.cpp

```diff
@@ -0,0 +1,18 @@
+#include "mlir/Support/LLVM.h"
+#include "triton/Dialect/Triton/IR/Dialect.h"
+
+namespace mlir::triton {
+
+SmallVector<NamedAttribute> getAllowedDiscardableAttrs(triton::AddPtrOp op) {
+  std::array<StringRef, 3> allowList{"tt.divisibility", "tt.contiguity",
+                                     "tt.constancy"};
+  SmallVector<NamedAttribute> propagatedAttrs;
+  for (auto attrName : allowList) {
+    Attribute attr = op->getDiscardableAttr(attrName);
+    if (attr)
+      propagatedAttrs.emplace_back(attrName, attr);
+  }
+  return propagatedAttrs;
```

Comment:
this is a bit of a nit but I don't think it should be about whether the attributes are "allowed" or not but whether the given transformation keeps the attributes correct. So I would prefer a local check that conservatively allow attributes to be propagated in the given pattern.
For instance, let's say we had a new attribute later on that wouldn't be valid to propagate in your pattern you don't want to it to be automatically propagated just because it is marked as allowed